spinalcordtoolbox/aggregate_slicewise.py

```diff
@@ -146,6 +146,33 @@ def func_map(data, mask, map_clusters):
     return beta[0], beta
 
 
+def func_median(data, mask=None, map_clusters=None):
+    """
+    Compute weighted median.
+    Code inspired from: https://gist.github.com/tinybike/d9ff1dad515b66cc0d87
+    :param data: nd-array: input data
+    :param mask: (n+1)d-array: input mask
+    :param map_clusters: not used
+    :return:
+    """
+    # Check if mask has an additional dimension (in case it is a label). If so, select the first label
+    if mask.ndim == data.ndim + 1:
+        mask = mask[..., 0]
+    data, mask = data.reshape(-1), mask.reshape(-1)
+    s_data, s_mask = map(np.array, zip(*sorted(zip(data, mask))))
+    midpoint = 0.5 * sum(s_mask)
+    if any(mask > midpoint):
+        w_median = (data[mask == np.max(mask)])[0]
+    else:
+        cs_mask = np.cumsum(s_mask)
+        idx = np.where(cs_mask <= midpoint)[0][-1]
+        if cs_mask[idx] == midpoint:
+            w_median = np.mean(s_data[idx:idx + 2])
+        else:
+            w_median = s_data[idx + 1]
```

Comment:
I tried replacing this snippet with a small package called [`wquantiles`](https://github.com/nudomarinero/wquantiles) (I looked over the code and it seems well-tested + has additional input validation, so I thought it'd be nice for double-checking.) 

```suggestion
    w_median = wquantiles.median(data=data, weights=mask)
```

I got slightly different results! 

```
>       assert agg_metric[list(agg_metric)[0]]['MEDIAN()'] == 39.0
E       assert 39.5 == 39.0
```

Reply:
oopsi! that's a problem. OK, i need to investigate further.

Reply:
hum, so I downloaded `wquantiles` and tried this example from the definition of the weighted median on wikipedia and got 3.6 instead of the 4:
```python
>>> data = np.array([1, 2, 3, 4, 5])
>>> mask = np.array([0.15, 0.1, 0.2, 0.3, 0.25])
>>> wquantiles.median(data, mask)
3.6
```
ok... this is wikipedia, so to consider with a grain of salt... i continue my investigations...

Reply:
Trying a comparison of `wquantiles` with another Python library for weighted statistics: [robustats](https://github.com/FilippoBovo/robustats)
```
>>> x = np.array([1.1, 5.3, 3.7, 2.1, 7.0, 9.9])
>>> weights = np.array([1.1, 0.4, 2.1, 3.5, 1.2, 0.8])
>>> wquantiles.median(x, weights)                                          
3.0714285714285716  # ⚠️ should be # 2.1
```

Reply:
and the wikipedia example with robustats:
```python
>>> data = np.array([1, 2, 3, 4, 5])
>>> mask = np.array([0.15, 0.1, 0.2, 0.3, 0.25])
>>> robustats.weighted_median(data, mask)
4.0
```

Reply:
here is the explanation for the discrepancies: https://github.com/nudomarinero/wquantiles/issues/4

Reply:
OK, after giving it some thoughts, I think I prefer the non-discrete implementation of the median. So I will go with the **wquantiles** package. Thank you for finding it @joshuacwnewton, and for raising this debate over the definition of the median. I am better informed now 😅

Reply:
Thank YOU for taking the time to look deeply into this! 

One last nit: I notice you've updated the docstring of the function, but perhaps we should also be explicit about the chosen method in the argparse description for `-method median`, too.

Reply:
excellent point, fixed in https://github.com/neuropoly/spinalcordtoolbox/pull/3329/commits/af686d5532143b56a5b65f95a2dc0cbc57a494af